lib/handlers/tenderly-sim/tenderly-sim.ts

```diff
@@ -0,0 +1,122 @@
+import { JsonRpcProvider, TransactionReceipt } from '@ethersproject/providers'
+import { log } from '@uniswap/smart-order-router'
+import axios from 'axios'
+import { BigNumber, ethers, providers } from 'ethers'
+
+// Swap Router Contract
+const V3_ROUTER2_ADDRESS = '0x68b3465833fb72A70ecDF485E0e4C7bD8665Fc45'
+
+// API for GET+DELETE
+export const TENDERLY_FORK_API_URL = (FORK_ID: string): string => `https://rpc.tenderly.co/fork/${FORK_ID}`
+
+// API For POST
+export const POST_TENDERLY_FORK_API_URL = (
+  TENDERLY_BASE_URL: string,
+  TENDERLY_USER: string,
+  TENDERLY_PROJECT: string
+) => `${TENDERLY_BASE_URL}/api/v1/account/${TENDERLY_USER}/project/${TENDERLY_PROJECT}/fork`
+
+export const APPROVE_TOKEN_FOR_TRANSFER =
+  '0x095ea7b300000000000000000000000068b3465833fb72a70ecdf485e0e4c7bd8665fc45ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
+
+export interface ISimulator {
+  simulateTx: (
+    chainId: number,
+    hexData: string,
+    tokenInAddress: string,
+    fromAddress: string,
+    blockNumber: number
+  ) => Promise<TransactionReceipt>
+}
+export class TenderlyProvider implements ISimulator {
```

Comment:
We should move all this into SOR and make it a regular part of the business logic there. 

Simulate=true/false should be a flag in the `SwapOptions` type, and if its true we should override all the gas related fields that are returned to be the more accurate ones from the simulation.

This is a very useful feature that we should also expose to people that are using the SOR directly, and potentially to things like the widget that use the SOR client side.